### DISCUSSION TITLE
Common response wrappers

### DISCUSSION THREAD

**Discussion by acwazz** [litestar-org/litestar#910] [Q&A] [answered]
Hello, is there a way to wrap handlers responses (JSON/HTTP) with a generic schema? 
Can this feature work also in the OpenAPI schema generation?

For example, an endpoint:
```py
from dataclasses import dataclass
from typing import Generic, TypeVar
from starlite import get
from pydantic import BaseModel

T = TypeVar("T")

@dataclass()
class ResponseWrapper(Generic[T]):
    data: T
    error: bool = False

# something to tell Starlite to wrap some (or all) HTTP handlers responses with a `ResponseWrapper`
...

class Pottery(BaseModel):
    name: str

@get("/pottery")
def retrieve_pottery() -> Pottery:
     return Pottery(name="Teapot")

# Registering the handler in Starlite
...
```

Returns a response like:
```jsonc
// GET /pottery 200 OK
{
    "error": false,  // Metadata
    "data": {"name": "Teapot"}  // Here the `Pottery` resource
}
```

Already tried to use generics in handlers return annotation (used like the pagination feature) but the OpenAPI schema generation broke.

**Comment by Goldziher** [comment, marked as answer]
Hi,

No there is no such a feature. You can use the `after_request` hook to transform responses into this form, and you can also define a custom `exception_handler`, so you can basically do this.

**Reply by acwazz** [reply]
Ok cool, thank you!
